public/app/features/explore/utils/logsVolumeResponse.ts

```diff
@@ -0,0 +1,19 @@
+import { DataQueryError, DataQueryResponse } from '@grafana/data';
+
+// Currently we can only infer if an error response is a timeout or not.
+// TODO: add backend support for specifying the type of error.
+export function isTimeoutErrorResponse(response: DataQueryResponse | undefined): boolean {
+  if (!response) {
+    return false;
+  }
+  if (!response?.error && !response?.errors) {
+    return false;
+  }
+
+  const errors = response?.error ? [response?.error] : response?.errors || [];
+
+  return errors.reduce((isTimeoutError: boolean, error: DataQueryError) => {
+    const message = `${error.message || error.data?.message}`?.toLowerCase();
+    return isTimeoutError || message.includes('timeout');
+  }, false);
```

Comment:
`DataQueryError` has an optional `type` field that is defined as:

```
export enum DataQueryErrorType {
  Cancelled = 'cancelled',
  Timeout = 'timeout',
  Unknown = 'unknown',
}
```

why are we not using that instead?

if we can't can we avoid using a reduce here? `some should be sufficient:
```suggestion
  return errors.some((error) => {
    const message = `${error.message || error.data?.message}`?.toLowerCase();
    return message.includes('timeout');
  });
```